spec/factories/generic_works.rb

```diff
@@ -88,31 +88,95 @@
     end
 
     factory :with_embargo_date do
+      # build with defaults:
+      # let(:work) { create(:embargoed_work) }
+
+      # build with specific values:
+      # let(:embargo_attributes) do
+      #   { embargo_date: Date.tomorrow.to_s,
+      #     current_state: Hydra::AccessControls::AccessRight::VISIBILITY_TEXT_VALUE_PRIVATE,
+      #     future_state: Hydra::AccessControls::AccessRight::VISIBILITY_TEXT_VALUE_PUBLIC }
+      # end
+      # let(:work) { create(:embargoed_work, with_embargo_attributes: embargo_attributes) }
+
       transient do
+        with_embargo_attributes false
         embargo_date { Date.tomorrow.to_s }
         current_state { Hydra::AccessControls::AccessRight::VISIBILITY_TEXT_VALUE_PRIVATE }
         future_state { Hydra::AccessControls::AccessRight::VISIBILITY_TEXT_VALUE_PUBLIC }
       end
       factory :embargoed_work do
-        after(:build) { |work, evaluator| work.apply_embargo(evaluator.embargo_date, evaluator.current_state, evaluator.future_state) }
+        after(:build) do |work, evaluator|
+          if evaluator.with_embargo_attributes
+            work.apply_embargo(evaluator.with_embargo_attributes[:embargo_date],
+                               evaluator.with_embargo_attributes[:current_state],
+                               evaluator.with_embargo_attributes[:future_state])
+          else
+            work.apply_embargo(evaluator.embargo_date,
+                               evaluator.current_state,
+                               evaluator.future_state)
+          end
+        end
       end
       factory :embargoed_work_with_files do
-        after(:build) { |work, evaluator| work.apply_embargo(evaluator.embargo_date, evaluator.current_state, evaluator.future_state) }
+        after(:build) do |work, evaluator|
```

Comment:
If this is doing the same thing as the factory above, how about pulling it up into the parent factory `:with_embargo_date` to DRY it up?  It looks like the same could be done for the `:with_lease_date` factory below.

Reply:
@laritakr and I chatted and agreed that my suggestion is a refactor that is outside the scope of this PR.